lib/api.js

```diff
@@ -282,117 +264,77 @@ api.declare({
   return res.reply(result);
 });
 
-function createPubKeyHash(pubKey) {
-  assert(typeof pubKey === 'string');
-  let keyData = pubKey.split(' ');
-  assert(keyData.length >= 2, 'pub key must be in a valid format');
-  keyData = keyData[0] + ' ' + keyData[1];
-  keyData = crypto.createHash('sha256').update(keyData).digest('hex');
-  return keyData.slice(0, 7);
-};
-
-function createKeyPairName(prefix, pubKey, workerName) {
-  assert(typeof prefix === 'string');
-  // We want to support the case where we're still using a config setting
-  // that ends in : as it used to
-  if (prefix.charAt(prefix.length - 1) === ':') {
-    prefix = prefix.slice(0, prefix.length - 1);
-  }
-  assert(prefix.indexOf(':') === -1, 'only up to one trailing colon allowed');
-  assert(typeof pubKey === 'string');
-  assert(typeof workerName === 'string');
-  return prefix + ':' + workerName + ':' + createPubKeyHash(pubKey);
-};
-
-function parseKeyPairName(name) {
-  assert(typeof name === 'string');
-  let parts = name.split(':');
-  assert(parts.length === 3, 'Unparsable keypair name: ' + name);
-  return {
-    prefix: parts[0],
-    workerType: parts[1],
-    keyHash: parts[2],
-  };
-};
-
 api.declare({
   method: 'get',
-  route: '/worker-types/:workerType/key-pair',
```

Comment:
According to the logs, the provisioner calls this endpoint quite a bit, so changing the path will likely cause those calls to fail.

Reply:
yep, which is the provisioner patch that I was mentioning in the original message.

Reply:
github is showing me comments out of order.

Wander, since the API is declared Experimental and the only known consumer is receiving a patch at the same time, there's no reason to do a staged deployment here.